### PR TITLE
chore(js-v4): updates readme examples

### PR DESCRIPTION
**Pull request by nunomaduro**
This pull request updates the readme for the JavaScript v4 client. Careful, I am not sure why there was a `callback` based API on the `getUserToken` method.

Just merge this once JSV4 is out.

### CODE REVIEW

**Comment by nunomaduro** [comment]
@tkrugg You mean the user needs the call the `init` before correct?

**Comment by nunomaduro** [comment]
@Haroenv Correct, I am aware.

File: README.md [hunk 1]
```diff
@@ -185,13 +185,15 @@ const search = instantsearch({
 
 #### Access `userToken`
 
-In cases where the `userToken` is generated, you need a way to access the `userToken` so that you can pass it to the `searchClient`.
+In cases where the `userToken` is generated, you should get the `UserToken` and pass it to the `searchClient`.
 
-```ts
-const searchClient = algoliasearch('APPLICATION_ID', 'SEARCH_API_KEY');
+```js
+const userToken = aa('getUserToken');
```
Comment (Haroenv) [inline README.md]: this isn't synchronous, so the flow described here isn't possible :)

**Comment by nunomaduro** [comment]
This solution will work for now, ideally, once this pull request is merged: https://github.com/algolia/doc/pull/4042. We would update this snippet to: 

```js
const searchClient = algoliasearch('APPLICATION_ID', 'SEARCH_API_KEY');

const searchOptions = {};
const userToken = aa('getUserToken', null, (err, userToken) => {
  searchOptions['userToken'] = userToken;
});

// Make sure you use `searchOptions` in search related operations
searchClient
  .initIndex('INDEX_NAME')
  .search('query', searchOptions)
  .then(({ hits }) => {
    console.log(hits);
  });
```

**Comment by nunomaduro** [comment]
@Haroenv done

**Comment by nunomaduro** [comment]
@Haroenv Thanks. Let's merge this once the client is out.

**Pull request merged**